app/src/main/java/app/wefridge/wefridge/model/ItemController.kt

```diff
@@ -145,8 +144,9 @@ class ItemController {
 
                         for (documentSnapshot in snapshots.documentChanges) {
                             with(documentSnapshot) {
-                                val item = parse(document)
-                                listener(item, type, oldIndex, newIndex)
+                                val item = tryParse(document)
+                                if (item != null)
+                                    listener(item, type, oldIndex, newIndex)
```

Comment:
I don't think that this a good idea, if only one listener() is not called on a documentChange, the whole fragment could get out of sync.
```kt
val item = tryParse(document)
listener(item, type, oldIndex, newIndex)
```
and then change the signature to
```kt
listener: (Item?, DocumentChange.Type, Int, Int) -> kotlin.Unit
```
and in PantryFragment
```kt
                DocumentChange.Type.ADDED -> {
                    values.add(newIndex, item!!)
                    recyclerViewAdapter.notifyItemInserted(newIndex)
                }
                DocumentChange.Type.MODIFIED -> {
                    values[newIndex] = item!!
                    recyclerViewAdapter.notifyItemChanged(newIndex)
                }
```

Reply:
Okay, I will change that!